Description: systemd: Support custom ntp groups
 The ntp user can have a different group than "ntp", which would happen
 either through customization or because the system has private groups
 disabled.  The init script already supports this.
Forwarded: no
Author: Richard Laager <rlaager@wiktel.com>
Last-Update: 2017-11-08
--- a/etc/ntpd.service
+++ b/etc/ntpd.service
@@ -14,7 +14,7 @@
 # the clock on boot.  However, if Restart=yes is set, a malicious (or broken)
 # server could send the incorrect time, trip the panic threshold, and when
 # ntpd restarts, serve it the incorrect time (which would be accepted).
-ExecStart=/bin/sh -c 'exec /usr/sbin/ntpd -g -N -u ntp:ntp $(if [ /var/lib/ntp/ntp.conf.dhcp -nt /etc/ntp.conf ] ; then echo "-c /var/lib/ntp/ntp.conf.dhcp" ; fi)'
+ExecStart=/bin/sh -c 'exec /usr/sbin/ntpd -g -N $(getent passwd ntp | cut -f 3,4 -d:) $(if [ /var/lib/ntp/ntp.conf.dhcp -nt /etc/ntp.conf ] ; then echo "-c /var/lib/ntp/ntp.conf.dhcp" ; fi)'
 Restart=no
 
 [Install]
